secret/secret.go

```diff
@@ -16,6 +16,27 @@ import (
 	"github.com/mailgun/metrics"
 )
 
+// Secret is an interface for encrypting/decrypting and authenticating messages.
+type Secret interface {
+	// Seal takes a plaintext message and returns an encrypted and authenticated ciphertext.
+	Seal([]byte) (Sealed, error)
+
+	// SealWithKey is the same as Seal but a different key can be passed in.
+	SealWithKey([]byte, *[SecretKeyLength]byte) (Sealed, error)
+
+	// Open authenticates the ciphertext and, if it is valid, decrypts and returns plaintext.
+	Open(Sealed) ([]byte, error)
+
+	// OpenWithKey is the same as Open but a different key can be passed in.
+	OpenWithKey(Sealed, *[SecretKeyLength]byte) ([]byte, error)
+}
+
+// Sealed respresents an encrypted and authenticated message.
+type Sealed interface {
+	GetCiphertext() []byte
+	GetNonce() []byte
```

Comment:
Actually, if I do this, method names will clash with field names in `SealedBytes` which is not allowed. I can either make fields private and introduce a `NewSealedBytes` function or keep `Get*`  getter names in the interface. What do you prefer?